src/main/scala/com/exasol/cloudetl/kinesis/KinesisShardsMetadataReader.scala

```diff
@@ -0,0 +1,66 @@
+package com.exasol.cloudetl.kinesis
+
+import scala.collection.JavaConverters._
+
+import com.amazonaws.auth._
+import com.amazonaws.services.kinesis.{AmazonKinesis, AmazonKinesisClientBuilder}
+import com.amazonaws.services.kinesis.model._
+import com.exasol._
+import java.util
+
+object KinesisShardsMetadataReader {
+  def run(exaMetadata: ExaMetadata, exaIterator: ExaIterator): Unit = {
+    val kinesisArgumentsReader = new KinesisArgumentsReader(exaIterator)
+    val awsAccessKeyId = kinesisArgumentsReader.getAwsAccessKeyArgument
+    val awsSecretAccessKey = kinesisArgumentsReader.getAwsSecretKeyArgument
+    val awsSessionToken = kinesisArgumentsReader.getAwsSessionTokenArgument
+    val streamName = kinesisArgumentsReader.getStreamNameArgument
+    val region = kinesisArgumentsReader.getRegionArgument
+    val awsCredentials =
+      new BasicSessionCredentials(awsAccessKeyId, awsSecretAccessKey, awsSessionToken)
+    val amazonKinesis = AmazonKinesisClientBuilder.standard
+      .withRegion(region)
+      .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
+      .build
+    val shards = getAllShardsFromStream(streamName, amazonKinesis)
+    val kinesisMetadata = collectShardsIds(shards)
+    try {
+      kinesisMetadata.foreach(
+        shardId => exaIterator.emit(shardId)
+      )
+    } catch {
```

Comment:
Check if you need to close the amazonKinesis, if so, do it in a finally clause.

Reply:
done